docker/Dockerfile

```diff
@@ -0,0 +1,77 @@
+FROM ubuntu:20.04
+
+# Necessary to install tzdata. It will default to UTC.
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y \
+      python3.9 \
+      # python3.9-distutils is required to install pip
+      # it unfortunately has to install python3.8-minimal as well...
+      python3.9-distutils \
+      # python-is-python3 creates a symlink for python to python3
+      python-is-python3 \
+      # For creating virtual environments
+      python3.9-venv \
+      # wget is used to obtain pip
+      wget \
+      # apache
+      apache2 \
+      apache2-dev \
+      libapr1-dev \
+      apache2-utils \
+      sudo && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set python3 to python3.9 (otherwise, it will be python3.8)
+RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1
+
+# Install and upgrade pip
+RUN wget -O- https://bootstrap.pypa.io/get-pip.py | python3.9 && \
+    pip install -U pip
+
+# Install setup dependencies
+RUN pip install PyYAML
+
+# Set up needed permissions and users
+RUN groupadd proxy-mapping && \
+    groupadd trame-user && \
+    useradd --system -g trame-user -G proxy-mapping -s /sbin/nologin trame-user && \
+    usermod -a -G proxy-mapping www-data && \
+    useradd admin && echo "admin:admin" | chpasswd && adduser admin sudo && \
+    mkdir -p /opt/trame/log && \
+    chown -R trame-user:trame-user /opt/trame && \
```

Comment:
wondering is we need `mkdir -p /deploy/server` so it can work when we do `-v /.../:/deploy/server`.

Reply:
I gave the mounting a try without the `mkdir -p /deploy/server`, and it seems to work. We might be able to skip this.

Reply:
yes I was not sure since I did not try